pages/fringe-events.jsx

```diff
@@ -1,21 +1,247 @@
+    var schema = this.getSchema();
+    return (
+      <form action="/add-fringe-event" method="POST" className="mozfest-form">
+        {
+          Object.keys(schema).map(function(formName) {
+            var props = schema[formName];
+            schema[formName].name = formName;
+            return (
+              <Input {...props} key={formName}>{formName}</Input>
+            );
+          })
+        }
+        <fieldset>
+          <a onClick={this.submitForm} ref="submitButton" className="button centered" id="fringe-submit-button"><span>Submit</span></a>
+        </fieldset>
+      </form>
+    );
+  }
+});
+
+var FringePage = React.createClass({
+  getInitialState: function() {
+    return {
+      events: [],
+      eventsLoaded: false,
+      unableToLoadEvents: false
+    }
+  },
+  componentWillMount: function() {
+    this.getFringeEvents();
+  },
+  getFringeEvents: function() {
+    var self = this;
+    fetch('/get-fringe-events', {
+      method: 'get'
+    }).then(function(response) {
+      return response.json();
+    }).then(function(data) {
+      self.setState({ 
+        events: data, 
+        eventsLoaded: true,
+        unableToLoadEvents: false
+      });
+    }).catch(function(error) {
+      console.log("Error: ", error);
+      self.setState({ 
+        eventsLoaded: true,
+        unableToLoadEvents: true
+      });
+    });
+  },
   render: function() {
     return (
-      <div>
+      <div className="fringe-events-page">
         <Header/>
         <HeroUnit image="/assets/images/fringe.jpg"
-                  image2x="/assets/images/fringe.jpg"/>
-
+                  image2x="/assets/images/fringe.jpg">
+          Fringe Events
+        </HeroUnit>
+        <div className="white-background">
+          <div className="content centered wide">
+            <h1>If MozFest wasn’t enough for you&hellip;</h1>
+            <p>MozFest is about working in the open and sharing ideas with the greater community. There are several innovative events taking place around the world with this same philosophy, and we would love to align them with MozFest.</p>
+            <p>If you are running an aligned event independent of the festival, please fill out our Fringe Event form. We will share your event details on our Fringe Events page.</p>
+            <p>How do you know if your event is a Fringe Event? It can be held anywhere in the world, but must align with the MozFest ethos: open, collaborative workshops or discussions focused on building tools and resources to keep the Web free and innovative.</p>
+            <div className="cta">
+              <h2>Be part of the Fringe scene and list your event!</h2>
+              <a className="button" href="#fringe-form-section"><span>Submit Your Fringe Event</span></a>
+            </div>
+            <div className="horizontal-rule"></div>
+          </div>
+        </div>
+        <div className="events">
+          <ul>
+            {
+              !this.state.eventsLoaded ? 
```

Comment:
the first two cases look the same, except for the <p>, so it's better to collapse that